### PR TITLE
util for duration spoken seconds

### PR DESCRIPTION
**Pull request by benjamin-sylvanus**
Need to test more today but these shouldn't modify any existing behavior.

### CODE REVIEW

File: vocode/streaming/streaming_conversation.py [hunk 1]
```diff
@@ -1372,6 +1370,11 @@ def is_interrupt(self, transcription: Transcription):
             self.transcriber.get_transcriber_config().min_interrupt_confidence or 0
         )
 
+    def get_duration_spoken_seconds(self, time_started_speaking: float | None):
```
Comment (bluegoo192) [inline vocode/streaming/streaming_conversation.py:1373]: let's make this a standalone function so readers can assume it doesn't rely on the millions of flags in this class

File: vocode/streaming/streaming_conversation.py [hunk 2]
```diff
@@ -1465,7 +1472,9 @@ async def send_speech_to_output(
         # Added stop event check otherwise it will block other synthesis result tasks
         # even though we meant to cancel this one.
         sleep_interval = 0.1  # Mark last action every 0.1 seconds
-        remaining_sleep = total_time_sent
+        remaining_sleep = total_time_sent - self.get_duration_spoken_seconds(
```
Comment (bluegoo192) [inline vocode/streaming/streaming_conversation.py:1475]: this sure looks like it will affect behavior

**Review by bluegoo192 — APPROVED** [review]
are you sure this won't affect behavior? the code looks fine (just move that function), but I'm worried your description implies hidden complexity

approved to avoid blocking, but don't merge this blindly

File: vocode/streaming/streaming_conversation.py (hunk 2 above)
Comment (benjamin-sylvanus) [reply vocode/streaming/streaming_conversation.py:1475]: Depends on what your definition of behavior is. Yes we are changing that value. Will it affect our conversation mechanics enough to modify how our agents respond to the same human input. I don't think so but as I said in the description I need to test more and key word **_shouldn't_** modify existing behavior.

File: vocode/streaming/streaming_conversation.py (hunk 1 above)
Comment (benjamin-sylvanus) [reply vocode/streaming/streaming_conversation.py:1373]: I am going to just add a`@staticmethod` decorator.

**Pull request merged**